orchagent/intfsorch.cpp

```diff
@@ -225,66 +281,29 @@ void IntfsOrch::doTask(Consumer &consumer)
                 continue;
             }
 
-            auto it_intfs = m_syncdIntfses.find(alias);
-            if (it_intfs == m_syncdIntfses.end())
+            if (!vnet_name.empty())
             {
-                if (addRouterIntfs(vrf_id, port))
+                VNetOrch* vnet_orch = gDirectory.get<VNetOrch*>();
+                if (!vnet_orch->isVnetExists(vnet_name))
                 {
-                    IntfsEntry intfs_entry;
-                    intfs_entry.ref_count = 0;
-                    m_syncdIntfses[alias] = intfs_entry;
+                    it++;
+                    continue;
                 }
-                else
+                if (!vnet_orch->setIntf(alias, vnet_name, ip_prefix_in_key ? &ip_prefix : nullptr))
                 {
                     it++;
                     continue;
                 }
             }
-
-            vrf_id = port.m_vr_id;
-            if (!ip_prefix_in_key || m_syncdIntfses[alias].ip_addresses.count(ip_prefix))
-            {
-                /* Request to create router interface, no prefix present or Duplicate entry */
-                it = consumer.m_toSync.erase(it);
```

Comment:
This case is now differently handled by the new code. It just erased and continued but with new code, since the function is returning true,  it is inserted to `m_syncdIntfses` at line 307. This will insert a default `IpPrefix ` when there is no prefix present in key (VRF/VNET) case

Reply:
You're correct
Fixed